### PR TITLE
feat: 🚀 Action Runner Controller version 0.23.7

### PR DESCRIPTION
**Pull request by Tanveer143s**
## what
* Created Action-runner-controller addon code
* 
image
* Tested with self-hosted runner
Screenshot from 2024-02-15 20-36-52

### CODE REVIEW

File: addons/ingress-nginx/config/ingress_nginx.yaml [hunk 1]
```diff
@@ -449,7 +449,7 @@ controller:
       # -- Enable the external controller service or not. Useful for internal-only deployments.
       enabled: true
     # -- Annotations to be added to the external controller service. See `controller.service.internal.annotations` for annotations to be added to the internal controller service.
-    annotations: {}
+    annotations:
```
[the remark this answers is not in the document]
Comment (Tanveer143s) [reply addons/ingress-nginx/config/ingress_nginx.yaml]: Done

**Review by VishwajitNagulkar — APPROVED** [review]
LGTM

**Review by AshutoshAM2002 — APPROVED** [review]
LGTM

**Review by nileshgadgi — APPROVED** [review]

File: addons/actions-runner-controller/variables.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+variable "helm_config" {
+  description = "Helm provider config for Actions Runner Controller"
+  type        = any
```
[the remark this answers is not in the document]
Comment (Tanveer143s) [reply addons/actions-runner-controller/variables.tf:3]: used in local.tf

File: .github/workflows/readme.yml [hunk 3]
```diff
@@ -36,7 +36,7 @@ jobs:
       - name: Generate TF Docs
         uses: terraform-docs/gh-actions@v1.0.0
         with:
-          working-dir: addons/cluster-autoscaler,addons/cert-manager,addons/ingress-nginx,addons/keda,addons/reloader,addons/kubeclarity,addons/external-dns
+          working-dir: addons/cluster-autoscaler,addons/cert-manager,addons/ingress-nginx,addons/keda,addons/reloader,addons/kubeclarity,addons/external-dns,addons/action-runner-controller
```
[the remark this answers is not in the document]
Comment (Tanveer143s) [reply .github/workflows/readme.yml]: fixed

**Pull request merged**